src/http.cpp

```diff
@@ -921,6 +921,10 @@ Request::handle_response_header(const asio::error_code& ec)
                 std::bind(&Request::handle_response_body, this, std::placeholders::_1, std::placeholders::_2));
         }
     }
+    else if (response_.status_code >= 400 and response_.status_code <= 500)
```

Comment:
response with no body should be properly handled regardless of the HTTP response code

Reply:
@aberaud I think the empty body aka just headers are well handled in the lines before as tested with [http_errors.cpp](https://github.com/savoirfairelinux/opendht/pull/450) !